packages/react-table/src/components/Table/base/types.tsx

```diff
@@ -171,6 +171,10 @@ export interface ThSelectType {
   isSelected: boolean;
   /** Flag indicating the select checkbox in the th is disabled */
   isHeaderSelectDisabled?: boolean;
+  /** Whether to disable the selection */
+  disable?: boolean;
```

Comment:
maybe isDisabled?

Reply:
After slack conversation - updating to `isDisabled` and opened #8419 to update `TdSelectType` to match as a breaking change